src/machine.rs

```diff
@@ -102,10 +152,21 @@ pub struct Evaluator<'tcx> {
     pub(crate) communicate: bool,
 
     pub(crate) file_handler: FileHandler,
+
+    /// Extra information needed for unwinding
+    /// We create this even in abort mode, so
+    /// that we can perform some basic validation
+    /// during panics
+    pub(crate) cached_data: CachedTypes<'tcx>,
```

Comment:
I'd prefer type caching to be introduced in a separate PR, so that review here can focus on the unwinding bits.